go.mod

```diff
@@ -44,13 +38,20 @@ require (
 	github.com/spf13/viper v1.4.0
 	github.com/stretchr/objx v0.2.0 // indirect
 	github.com/stretchr/testify v1.4.0
+	go.opencensus.io v0.22.2 // indirect
 	go.uber.org/goleak v0.0.0-00010101000000-000000000000
```

Comment:
i'm confused about why goleak still appears here in the `require` directive ..

.. in fact, attempting to consume the datamon api from another module, i'm noticing that i get a build error unless i duplicate the `replace` directive for goleak..

.. not something that needs to be resolved as part of this merge request, although it does intersect the overall dep mgmt. story, and i could use some help getting that sorted in a less hack way.

Reply:
Goleak is a test dependency I introduced a couple weeks ago. Pb with this repo is that the latest release is niw quite old with sime breaking api changes in master. I agree that we could pin the commit directly rather than using replace which as you rightly point out not honored when consumed by another module. I am fixing this part today

Reply:
done